core/src/main/java/org/apache/iceberg/util/SnapshotUtil.java

```diff
@@ -101,6 +101,42 @@ public static Snapshot oldestAncestor(Table table) {
     return ancestorsOf(start, lookup);
   }
 
+  /**
+   * Traverses the history of the table's current snapshot
```

Comment:
This has a slightly different behavior now and i'm not sure the name or the doc match. If I follow this correctly we return

1) The Oldest Ancestor before a timestamp
2) If no such ancestor exists, just the oldest ancestor
3) If no snapshots exist null.

What is the difference between 1 and 2 here? Shouldn't 2 always return 1 if possible?

Reply:
Shouldn't this be
1) The youngest ancestor before a given timestamp
2) The oldest ancestor for the entire table if no such ancestor exists before the given timestamp
3) Null otherwise?

Reply:
@RussellSpitzer Thanks for the reviewing.
I think your understanding is totally correct, in the first comment, 2 should always return 1 if it is possible
Last commit logic:
`    for (Snapshot snapshot : currentAncestors(table)) {
      if (snapshot.timestampMillis() < timestamp) {
        return current;
      }

      current = snapshot;
    }

    return current;`

I think last commit will behavior that instead of calling oldestAncestor at last:
1.The youngest ancestor before a given timestamp
2.The oldest ancestor for the entire table if no such ancestor exists before the given timestamp
3.Null otherwise

When we reach the return function, current is equal to the result of oldestAncestor. This is also some part I am confused from the @rdblue 's comment above, I don't think we need to call oldestAncestor when the loop(var current will point to oldestAncestor). Please correct me if I understand something wrong

Reply:
I have taken the idea from @rdblue's code and have named the method `oldestAncestorFrom`. 

Please let me know if we are planning to call `oldestAncestor` if `oldestAncestorFrom` fails to find the oldest snapshot (instead of throwing an exception).

Reply:
I have taken the idea from @rdblue's code and have named the method `oldestAncestorFrom`. 

Considering it is a utility method, please let me know if we are planning to call `oldestAncestor` or `throw IllegalStateException` if `oldestAncestorFrom` fails to find the oldest snapshot.